### ISSUE TITLE
Support for arbitrary shifted keycodes

### ISSUE THREAD

**Issue by zhengqunkoo** [david-janssen/kmonad#33]
By default, shifted keycodes mask the corresponding non-shifted keycodes. Is it possible to define shifted keycodes that do not correspond to the non-shifted keycodes?

For example, the `dvorak-programmer` layout emits the QWERTY `6` keypress as a non-shifted `=`, and a shifted `9`.

The obvious solution is to define the shift-keys as a layer-toggle to a separate layer, e.g. `LAYER shift`. However, this has the slight downside that key combinations that need shift keys, e.g. `C-S-left` and `C-S-tab`, no longer work, unless many aliases are defined.

Is there any other way to do this? Is there a plan to support aliasing arbitrary shifted keycodes with a new "shift" keyword (keywords like `TH` and `TN`)?

**Comment by david-janssen** [comment]
I had to read your message a few times but if I understand correctly you want a key that, unshifted, emits an `=`, and shifted emits a `9`.

If that is the case, you are right, that is currently not possible without complicated layer-trickery. Additionally, currently the logic of `kmonad` doesn't make this a simple fix at all, although it is actually an interesting idea.

I might like to add support for this, but I have some other stuff to wrap up first, and am already quite short of time for working on kmonad lately, but I will definitely keep this on the list of things to do. I suggest leaving this issue open until I do :)

**Comment by david-janssen** [comment]
So, I gave this some more thought, and I've decided against implementing this behavior for 2 reasons:

1. Arbitrary shifted keycodes requires a complete redesign of how modifier keys are handled. I think it is much cleaner to simply emit keys, and let the OS interpret them as they will.
2. The desired behavior is actually already achievable, as detailed above. Essentially layers will get you what you want, without having to redesign the concept of what a button is and how it does things.
